gin-redis/README.md

```diff
@@ -0,0 +1,203 @@
+```
+
+> **Since, we are on the docker image the Redis URL will be `myredis:6379`. This needs to be updated on the on line 18 in `redisConnect.go` file**
+
+### Let's start the Redis Instance
+Using the docker-compose file we will start our Redis instance:-
+```bash
+sudo docker run -p 6379:6379 -d --network keploy-network --name myredis redis
+```
+```bash
+docker build -t gin-app:1.0 .
+```
+
+### Capture the Testcases
+
+```shell
+keploy record -c "docker run -p 3001:3001 --name RediApp --network keploy-network gin-app:1.0"
+```
+
+To genereate testcases we just need to make some API calls. You can use [Postman](https://www.postman.com/), [Hoppscotch](https://hoppscotch.io/), or simply `curl`
+
+### 1. Request OTP
+
+```bash
+
+curl --location 'localhost:3001/api/getVerificationCode?email=something%40gmail.com'
+```
+
+this will return the signup response. 
+```
+{
+    "status": "true",
+    "message": "2121"
+}
+```
+
+**2. Verify OTP**
+
+```bash
+curl --location 'localhost:3001/api/getVerificationCode?email=something%40gmail.com' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "otp":"2121",
+    "email":"shivamsouravjha@gmail.com"
+}'
+
+```
+
+_Now, let's see the magic! 🪄💫_
+
+Now both these API calls were captured as a testcase and should be visible on the Keploy CLI. You should be seeing an app named `keploy folder` with the test cases we just captured and data mocks created.
+
+### Run the captured testcases
+
+Now that we have our testcase captured, run the test file.
+
+```shell
+keploy test -c "sudo docker run -p 3001:3001 --rm --net keploy-network --name ginRedisApp gin-app:1.0" --delay 10
+```
+
+So no need to setup dependencies like Redis, web-go locally or write mocks for your testing.
+
+**The application thinks it's talking to Redis 😄**
+
+We will get output something like this:
+
+![TestRun](./img/testrun.png)
+
+
+## Run app Natively on local machine
+
+Keploy can be installed on Linux directly and on Windows with the help of WSL. Based on your system archieture, install the keploy latest binary release
+
+**1. AMD Architecture**
+
+```shell
+curl --silent --location "https://github.com/keploy/keploy/releases/latest/download/keploy_linux_amd64.tar.gz" | tar xz -C /tmp
+
+sudo mkdir -p /usr/local/bin && sudo mv /tmp/keploy /usr/local/bin && keploy
+```
+
+<details>
+<summary> 2. ARM Architecture </summary>
+
+```shell
+curl --silent --location "https://github.com/keploy/keploy/releases/latest/download/keploy_linux_arm64.tar.gz" | tar xz -C /tmp
+
+sudo mkdir -p /usr/local/bin && sudo mv /tmp/keploy /usr/local/bin && keploy
+```
+
+</details>
+
+#### Let's start the Redis Instance
+
+Spin up your Redis container using
+
+```shell
+sudo docker run -p 6379:6379 -d --network keploy-network --name myredis redis
```

Comment:
do we need keploy-network here..? (Local)

Reply:
<img width="720" alt="image" src="https://github.com/keploy/samples-go/assets/60891544/246b8aa3-b387-4e32-8eb7-471c0883c4bc">
Without network when I try create a dokcer image and try recording I get this error

Reply:
won't need it for native linux